src/umpire/ResourceManager.cpp

```diff
@@ -540,76 +540,82 @@ void ResourceManager::memset(void* ptr, int value, std::size_t length)
 void*
 ResourceManager::reallocate(void* current_ptr, std::size_t new_size)
 {
-  UMPIRE_LOG(Debug, "(current_ptr=" << current_ptr << ", new_size=" << new_size << ")");
-
-  UMPIRE_REPLAY(
-      R"( "event": "reallocate", "payload": { "ptr": ")"
-      << current_ptr
-      << R"(", "size": )"
-      << new_size
-      << R"( })"
-  );
-
-  void* new_ptr;
-
-  if ( current_ptr == nullptr || new_size == 0 ) {
-    if (current_ptr != nullptr )
-      m_default_allocator->deallocate(current_ptr);
-
-    new_ptr = m_default_allocator->allocate(new_size);
-  } else {
-    auto& op_registry = op::MemoryOperationRegistry::getInstance();
+  Allocator allocator;
```

Comment:
This could be a strategy pointer rather than the object, and then you can just construct the object inline when you call reallocate?

Having uninitialized variables is bad.

Reply:
agreed.  that was ugly..  fixed